aquarium/src/cg/group4/view/AquariumScreen.java

```diff
@@ -0,0 +1,249 @@
+     */
+    public AquariumScreen() {
+
+        createCollectionObserver();
+        createLabelObserver();
+
+        cInitialSet = new HashSet<CollectibleRenderer>();
+        cDisplayRendererSet = new HashSet<CollectibleRenderer>();
+
+        cDisplaySet = new Collection("$$Nil");
+
+        cStyle = new GameSkin();
+        cStyle.createUIElements(720);
+        cStage = new Stage();
+
+        initStage();
+        initTooltipLabels();
+    }
+
+    /**
+     * Adds an observer for the collection update, which is received from the server.
+     */
+    public void createCollectionObserver() {
+        cCollectionObserver = new Observer() {
+            @Override
+            public void update(Observable o, Object arg) {
+                Collection collection = (Collection) arg;
+                Gdx.app.log(this.getClass().getName(), "Received Collection update: " + collection.toString());
+
+
+                if (collection.isEmpty()) {
+                    notifyUserEmptyContainerReceived();
+                }
+
+                for (Collectible c : collection) {
+                    addCollectibleRendererItem(c);
+                }
+
+                for (CollectibleRenderer cr : cInitialSet) {
+                    if (!isInDisplaySet(cr)) {
+                        cDisplayRendererSet.add(cr);
+                        cStage.addActor(cr.getActor());
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * Creates the observer for the label.
+     */
+    public void createLabelObserver() {
+        cLabelObserver = new Observer() {
+            @Override
+            public void update(final Observable o, final Object arg) {
+                Pair<String> pair = (Pair<String>) arg;
+
+                cOwnerLabel.setText("owner id: " + pair.getElement1());
+                cDateLabel.setText("date of achievement: " + pair.getElement2());
+            }
+        };
+    }
+
+    /**
+     * Checks whether the display set already contains the given collectible.
+     * @param collectibleRenderer check whether this element exists in the already displayed set.
+     * @return true if in the set, false if not.
+     */
+    public boolean isInDisplaySet(final CollectibleRenderer collectibleRenderer) {
+        if (cDisplayRendererSet.contains(collectibleRenderer)) {
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * For a single collectible renderer, add an object to the set which will be used to render.
+     * @param collectible
+     */
+    public void addCollectibleRendererItem(Collectible collectible) {
+        CollectibleRenderer collectibleRenderer = new CollectibleRenderer(collectible);
+        collectibleRenderer.getSubject().addObserver(cLabelObserver);
+        cInitialSet.add(collectibleRenderer);
+    }
+
+    public void notifyUserEmptyContainerReceived() {
+        cStatusLabel.setText("Received an empty collection from the server!");
```

Comment:
It sounds like this is a bug now, maybe just say: the collection for this group is empty?